graylog2-server/src/main/java/org/graylog2/rest/resources/system/logs/LoggersResource.java

```diff
@@ -151,6 +148,49 @@ public void setSingleLoggerLevel(
         logger.setLevel(newLevel);
     }
 
+    @GET
+    @Timed
+    @ApiOperation(value = "Get recent log messages")
+    @ApiResponses(value = {
+            @ApiResponse(code = 404, message = "Memory appender is disabled or of wrong instance.")
+    })
+    @Path("/messages/recent")
+    @Produces(MediaType.APPLICATION_JSON)
+    public LogMessagesSummary messages(@ApiParam(name = "limit", required = true) @QueryParam("limit") int limit) {
+        List<InternalLogMessage> messages = new ArrayList<>();
```

Comment:
This list could be initialized with the correct size (`appender.getLogMessages(limit).size()`).